nucypher/network/server.py

```diff
@@ -273,22 +257,17 @@ def revoke_arrangement(id_as_hex):
 
         revocation = Revocation.from_bytes(request.data)
         log.info("Received revocation: {} -- for arrangement {}".format(bytes(revocation).hex(), id_as_hex))
+
+        # Check that the request is the same for the provided revocation
+        if not id_as_hex == revocation.arrangement_id.hex():
+            log.debug("Couldn't identify an arrangement with id {}".format(id_as_hex))
+            return Response(status_code=400)
```

Comment:
Response message for context perhaps...`Mismatched arrangement ID`?

Reply:
I'll defer judgement here to @jMyles since these responses aren't in the scope of the PR, but more than happy to make them more specific.